src/Router.js

```diff
@@ -160,6 +161,10 @@ export default class Router {
 
       this.__currentCanonicalPath = canonicalPath
 
+      if (this.onRouteStart && mode !== 'replace') {
+        this.onRouteStart();
```

Comment:
Yea, good point, I'll pass `fromPath` and `startTime`. Could you clarify what you meant about redirects?